consensus/core/src/dag_state.rs

```diff
@@ -83,11 +83,30 @@ impl DagState {
     /// Accepts a block into DagState and keeps it in memory.
     pub(crate) fn accept_block(&mut self, block: VerifiedBlock) {
         let block_ref = block.reference();
+        // Ensure we don't write multiple unique blocks per slot for our own index
+        if block_ref.author == self.context.own_index {
+            let existing_blocks = self.get_uncommitted_blocks_at_slot(block_ref.into());
+            if let Some(existing_block) = existing_blocks.first() {
```

Comment:
Since invariants are checked, we may as well check if existing_blocks.is_empty(). I don't think Core will support re-creating the same block.